src/Helpers_TEST.cc

```diff
@@ -518,6 +518,79 @@ TEST(HelpersTest, Pair)
   }
 }
 
+/////////////////////////////////////////////////
+TEST(HelpersTest, timePointToSecNsec)
+{
+  std::pair<int64_t, int64_t> parts = math::timePointToSecNsec(
+      std::chrono::system_clock::from_time_t(0));
+  EXPECT_EQ(parts.first, 0);
+  EXPECT_EQ(parts.second, 0);
+
+  using std::chrono::duration_cast;
+  using std::chrono::nanoseconds;
+  using std::chrono::system_clock;
+  std::chrono::system_clock::time_point point =
+    std::chrono::system_clock::from_time_t(0);
+  point += duration_cast<system_clock::duration>(nanoseconds(1000));
+  parts = math::timePointToSecNsec(point);
+
+  EXPECT_EQ(parts.first, 0);
+  EXPECT_EQ(parts.second, 1000);
+
+  point = std::chrono::system_clock::from_time_t(0);
+  point += std::chrono::seconds(60);
+  point += duration_cast<system_clock::duration>(nanoseconds(5789));
+  parts = math::timePointToSecNsec(point);
+
+  EXPECT_EQ(parts.first, 60);
+  EXPECT_EQ(parts.second, 5789);
```

Comment:
this expectation is failing on macOS, it gets `5000` instead of `5789`. My guess is that the `system_clock::duration` cast in line 542 is changing the time base to microseconds, presumably because that's the resolution of the system clock?

I'm not sure what the best way to account for that is

Reply:
oops, I should have read @JShep1's comment first: https://github.com/ignitionrobotics/ign-math/pull/150#discussion_r479505267

Reply:
perhaps we should expect `duration_cast<system_clock::duration>(5789)`?